### PR TITLE
Add option to automatically install user requested packages during setup

### PR DESCRIPTION
**Pull request by tomrund**
I had to setup my TeslaUSB again this weekend because my "high endurance" 256GB SanDisk card went into read-only mode after 4 months of use. This would've been helpful to streamline things a bit.

### CODE REVIEW

File: setup/pi/setup-teslausb [hunk 1]
```diff
@@ -921,6 +921,13 @@ make_root_fs_readonly
 
 upgrade_packages
 
+if [ -n "${INSTALL_USER_REQUESTED_PACKAGES:-}" ]
+then
+  export INSTALL_USER_REQUESTED_PACKAGES
+  copy_script setup/pi/install-user-requested-packages.sh /tmp
+  /tmp/install-user-requested-packages.sh
```
Comment (marcone) [inline setup/pi/setup-teslausb]: I don't think it's worth having a whole separate install script just to call `apt-get`.
Just do it inline here.

Comment (tomrund) [reply setup/pi/setup-teslausb]: I was on the fence about this one, I almost made this change prior to your review. I'm out of the country for another week or so, when I return I'll get this fixed up, thank you.

File: setup/pi/setup-teslausb [hunk 2]
```diff
@@ -921,6 +921,13 @@ make_root_fs_readonly
 
 upgrade_packages
 
+if [ -n "${INSTALL_USER_REQUESTED_PACKAGES:-}" ]
+then
+  setup_progress "Installing user requested packages: ${INSTALL_USER_REQUESTED_PACKAGES}"
+  # shellcheck disable=SC2086
```
Comment (marcone) [inline setup/pi/setup-teslausb:927]: I think this "shellcheck disable" can can be avoided by using `eval "DEBIAN_FRONTEND=noninteractive apt-get -y --force-yes install $INSTALL_USER_REQUESTED_PACKAGES"`, which isn't pretty but I like it slightly better than disabling shellcheck checks.

Comment (marcone) [reply setup/pi/setup-teslausb:927]: a nicer way would be to expand the setup variable into an array (or make it an array to begin with) and then expanding the array.
Something like:
```
read -r -a PACKAGES <<< "${INSTALL_USER_REQUESTED_PACKAGES[@]}"
DEBIAN_FRONTEND=noninteractive apt-get -y --force-yes install "${PACKAGES[@]}"
```
should work for both both a simple space-separated list as well as a proper array.

Comment (marcone) [reply setup/pi/setup-teslausb:927]: I've updated the change accordingly and merged it.

**Comment by marcone** [comment]
Merged modified version of this: https://github.com/marcone/teslausb/commit/7382e28279141220b8f6acce3ba38d23a9b6b170

File: setup/pi/setup-teslausb (hunk 2 above)
Comment (tomrund) [reply setup/pi/setup-teslausb:927]: Thank you @marcone.

**Pull request closed without merging**